File: addToArrayForm.java
import java.util.*;
// Approach 1- School book maths
class Solution { // TC O(n) SC O(n)
    public List<Integer> addToArrayForm(int[] num, int k) {
        List<Integer> ans=new ArrayList<>();
        int n=num.length;
        int curr=k;
        int i=n;
        while(i-- >=1 || curr>0){
            if(i>=0){
                curr+=num[i];
            }
            ans.add(curr%10);
            curr/=10;
        }
        Collections.reverse(ans);
        return ans;
    }
}

class addToArrayForm{
    public static void main(String[] args) {
        int[] num={1,2,3,4};
        int k=98;
        Solution s= new Solution();
        System.out.println(s.addToArrayForm(num,k));
    }
}

File: addToArrayForm.py
# Approach 1- School book maths
def addToArrayForm(num,k): # TC O(n) // SC O(n)
	ans=[]
	curr=k
	i=len(num)-1
	while i>=0 or curr>0:
	    if i>=0:
	        curr+=num[i]
	    ans.append(curr%10)
	    curr//=10
	    i-=1
	ans.reverse()
	return ans

print(addToArrayForm([1,2,3,4],56))